test/evaluation/filterShouldExistIfPredicateEvaluationHandler.spec.ts

```diff
@@ -0,0 +1,24 @@
+import fs from 'fs';
+
+import { PresentationDefinition } from '@sphereon/pe-models';
+
+import { EvaluationClient } from "../../lib/evaluation/evaluationClient";
+
+function getFile(path: string) {
+    return JSON.parse(fs.readFileSync(path, 'utf-8'));
+}
+
+describe('evaluate', () => {
+
+    it('should report error if the predicate is present and filter is not', function () {
+        const pdSchema: PresentationDefinition = getFile('./test/dif_pe_examples/pd/pd-simple-schema-age-predicate.json').presentation_definition;
+        delete pdSchema.input_descriptors[0].constraints.fields[0].filter;
+        const vpSimple = getFile('./test/dif_pe_examples/vp/vp-simple-age-predicate.json');
+        const evaluationClient: EvaluationClient = new EvaluationClient();
```

Comment:
I suggest to test each evaluation handler independently and have a separate test for the chain rather than testing the whole chain in every test. The way it is, if one test in the beginning of the chain fails they will cascade and it will hard to find out where the problem is.